tests_pytest/common_tests/unit_tests/core/test_fusion_info.py

```diff
@@ -0,0 +1,186 @@
+#  Copyright 2025 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#  ==============================================================================
+
+import pytest
+from unittest.mock import Mock
+
+from model_compression_toolkit.core.common.fusion.fusing_info import FusingInfoGenerator, FUSED_OP_ID_PREFIX
+from model_compression_toolkit.target_platform_capabilities import FrameworkQuantizationCapabilities
+from model_compression_toolkit.core.common import BaseNode
+
+
+class MockBaseNode:
+    """
+    Mock implementation of a base graph node.
+    Allows for equality checks and hashing based on the node name.
+    """
+
+    def __init__(self, name: str):
+        self.name = name
+
+    def __eq__(self, other):
+        return isinstance(other, MockBaseNode) and self.name == other.name
+
+    def __hash__(self):
+        return hash(self.name)
+
+
+@pytest.fixture
+def fusing_patterns():
+    """
+    - Returns predefined fusing patterns: Conv2D + ReLU and Linear + Softmax.
+    """
+    return [["Conv2d", "ReLU"], ["Linear", "Softmax"]]
+
+
+@pytest.fixture
+def mock_nodes():
+    """
+    Creates mock nodes representing a simple neural network structure.
+    - Nodes: Conv2D, ReLU, Linear, Softmax.
+    """
+    node1 = Mock(spec=BaseNode)
+    node1.name = "conv"
+    node1.layer_class = "Conv2d"
+
+    node2 = Mock(spec=BaseNode)
+    node2.name = "relu"
+    node2.layer_class = "ReLU"
+
+    node3 = Mock(spec=BaseNode)
+    node3.name = "linear"
+    node3.layer_class = "Linear"
+
+    node4 = Mock(spec=BaseNode)
+    node4.name = "softmax"
+    node4.layer_class = "Softmax"
+
+    return [node1, node2, node3, node4]
+
+
+@pytest.fixture
+def mock_graph(mock_nodes):
+    """
+    Creates a mock graph with topologically sorted nodes and defined connectivity.
+    - Implements `get_next_nodes` and `get_prev_nodes` to maintain linear order.
+    """
+    graph = Mock()
+    graph.nodes.return_value = mock_nodes
+    graph.get_topo_sorted_nodes.return_value = mock_nodes
+
+    def get_next_nodes(node):
+        if node in mock_nodes:
+            index = mock_nodes.index(node)
+            return [mock_nodes[index + 1]] if index < len(mock_nodes) - 1 else []
```

Comment:
s it possible do define a more explicit behavior in the mock? it's difficult to follow what is the behavior you expect to get here